docs/advanced/styling.md

```diff
@@ -8,20 +8,337 @@ has_toc: false
 
 # Styling Tasks
 
-Each task entry has CSS styles that allow you to change the look and feel of how the tasks are displayed. The
-following styles are available.
+In rendered queries and Reading View, the Tasks plugin adds detailed CSS classes that represent many of each task's content, to allow for very extensive styling options via CSS.
+Not only each component in a rendered task line is tagged with classes to differentiate it, many components also add classes that represent the actual content of the task, so CSS rules can refer to data such as the relative due date of a task or its specific priority.
+
+## Basic Task Structure
+
+{: .released }
+The following description relates to a restructuring of the rendered tasks that was introduced in Tasks X.Y.Z.
+
+The Tasks plugin renders a task in the following structure (this refers to query results, but the Reading View is the same exact the top-most containers):
+
+```markdown
+- Obsidian code block (div class="block-language-tasks")
+  - Results list (ul class="plugin-tasks-query-result")
+    - Task (li class="task-list-item" + specific classes like "tasks-tag-atHome tasks-priority-medium tasks-due-past-1d")
+      - Task checkbox (li class="task-list-item-checkbox")
```

Comment:
They should definitely be specific classes, I'll merge from master and add them.

Reply:
I played around with custom statuses, and except adding them to this document under the Task `li` element, I think there's nothing left to do.
They are already represented in a way addressable by CSS with a data attribute and it's also extensively documented. (I'm personally not fond of the name `data-task`, and would have rather have it called `data-task-status`, but I guess it's way too late for that...)

Reply:
P.S: in the past days I started having regrets not implementing some of this PR as data attributes rather than classes, but I think on one wants to delay this PR further. Wish I thought of that several weeks ago!

Reply:
Thank you for testing this.

I'm not qualified to comment on data vs classes... And I imagine you are keen to see the back of this work!

But there is no pressure if you did want to try something different... Equally, fine to stick with it as it is.